M3-1628 Create Linode From Backup Maintenance

martinmckenna opened the pull request:
### Purpose

Fix some regressions in the _create from backup_ flow

### Notes
* First, I added copy to the _create from backup_ form to make clear to the user that the newly provisioned Linode must be manually booted up, as we do not have the ability to do this with APIv4.
* I also abstracted the `safeGetImageLabel` component for some uniformity between how image labels appeared on the LinodeDetail and LinodeLanding pages.
* Finally, I added a `linode_create` event to the whitelist because the a Linode that is created from a backup would otherwise never update, since there is no `linode_boot` event

johnwcallahan commented on src/features/linodes/LinodesCreate/LinodesCreate.tsx:
```diff
@@ -215,7 +215,9 @@ export class LinodeCreate extends React.Component<CombinedProps, State> {
               text: `This newly created Linode wil be created with
-                      the same password and SSH Keys (if any) as the original Linode`,
+                      the same password and SSH Keys (if any) as the original Linode.
+                      Also note that this Linode will need to be manual booted after it finishes
```
I think this should be "manually booted"

nwisnewski replied:
Will in the first sentence is misspelled 'This newly created Linode wil', and the second sentence is lacking a period.

Jskobos commented:
Still a few leftover weirdnesses here: when a Linode is created from backup, after provisioning you get the situation below (temporarily):

![screen shot 2018-10-24 at 9 29 56 am](https://user-images.githubusercontent.com/1624067/47433999-97079b80-d76f-11e8-8cde-a69864b40e40.png)

The image is reported as No Image until the Linode has booted. Not sure there's anything we can do about that one.

Re: the mini-spinner, the idea works well. Is it possible to adjust the spinner size, or use `Loading image...` or whatever? It's slightly weird to see the size of the panel adjust every time the tab is loaded.

nwisnewski commented:
When creating a new linode from a backup, the provisioning loading indicator completes on the new linode, but the final status returned from `GET /v4/linode/instances/{id}` is "restoring", and there is no status indicator displayed in the Linodes page view for the new linode. If you attempt to then reboot when there is no status indicator you get 400 on the `POST /v4/linode/instances/{id}/reboot` call and an error message is displayed. If you navigate away from Linodes view then back,  or refresh the page, you'll then see the powered off red status on the linode created from backup. So maybe there need to be more polling when status is "restoring", or an additional status icon to prevent the reboot error

martinmckenna commented:
@pat @nwisnewski Addressed your feedback. There's still some wonkiness due to the fact that there is two events with progress happening at the same time (`backups_restore` and `linode_create`), so the component jumps back and forth between showing progress for both.

I'm going to create a ticket to create a pattern for handling simultaneous events with progress
